src/Compilers/Core/Portable/SymbolDisplay/AbstractSymbolDisplayVisitor.cs

```diff
@@ -75,6 +68,53 @@ protected AbstractSymbolDisplayVisitor NotFirstVisitorNamespaceOrType
             }
         }
 
+        protected virtual void Allocate(
+            ArrayBuilder<SymbolDisplayPart> builder,
+            SymbolDisplayFormat format,
+            bool isFirstSymbolVisited,
+            SemanticModel? semanticModelOpt,
+            int positionOpt,
+            bool inNamespaceOrType)
+        {
+            Debug.Assert(format != null);
+
+            if (_format is not null)
+                throw new InvalidOperationException();
```

Comment:
replace with `Contract.ThrowIfTrue(_format is not null)`?

Reply:
Is `Contract` available in the compiler layer?

Reply:
➡️ Contract is not available here